### ISSUE TITLE
failed upgrade to Omero5.5 - Ice 3.6 not found

### ISSUE THREAD

**Issue by vncntprvst** [openmicroscopy/openmicroscopy#6199]
Hi,
I've tried upgrading my Omero server to the 5.5 version, and I kept getting that `ERROR: Ice version mismatch!` from the init file in the OMERO.server/lib/python directory, which prevented using any omero command. All the prerequisites including a proper Ice 3.6 install are there, but that file kept finding Ice 3.5 as the installed version (I later uninstalled it completely but that didn't help). Interestingly, running that same check in ipython doesn't throw that error:
```
In [1]: import Ice
In [2]: vers = Ice.stringVersion()
Out[3]: '3.6.5'
```
So the init script is getting confused somehow. Just to see what would happen, I commented the lines for the version check and it did allow me to setup a new database, but now is stuck again, throwing this error:

> ERROR: Could not import omero.cli! (No module named Ice)
...
>  Debugging Info:
        --------------
        CWD=/home/omero/OMERO.server-5.5.1-ice36-b122
        VERSION=2.7.12 (default, Dec  4 2017, 14:50:18)  [GCC 5.4.0 20160609]
        OMERO_HOME=None
        PYTHONPATH=['/home/omero/OMERO.server/lib/python', '/home/omero/OMERO.server/var/lib', '/home/omero/OMERO.server-5.5.1-ice36-b122/bin', '/usr/lib/python2.7', '/usr/lib/python2.7/plat-x86_64-linux-gnu', '/usr/lib/python2.7/lib-tk', '/usr/lib/python2.7/lib-old', '/usr/lib/python2.7/lib-dynload', '/usr/local/lib/python2.7/dist-packages', '/usr/lib/python2.7/dist-packages', '/usr/lib/python2.7/dist-packages/PILcompat', '/usr/lib/python2.7/dist-packages/gtk-2.0', '/usr/lib/python2.7/dist-packages/ubuntu-sso-client', '/home/omero/OMERO.server/lib/fallback']

I'm at a loss about what's going on here.

Here's what the diagnostics returns (the init file restored as it was): 

> ================================================================================
OMERO Diagnostics (admin) 5.5.1-ice36-b122
================================================================================

> 
> Commands:   java -version                  11.0.5    (/usr/bin/java)
> Commands:   python -V                      2.7.12    (/usr/bin/python)
> Commands:   icegridnode --version          3.6.5     (/usr/bin/icegridnode)
> Commands:   icegridadmin --version         3.6.5     (/usr/bin/icegridadmin)
> Commands:   psql --version                 11.6      (/usr/bin/psql)
> Commands:   openssl version                1.0.2     (/usr/bin/openssl)
> 
> Server:     icegridnode                    not started
> 
> Log dir:    /home/omero/OMERO.server/var/log
> No logs available
> 
> Environment:OMERO_HOME=(unset)
> Environment:OMERO_NODE=(unset)
> Environment:OMERO_MASTER=(unset)
> Environment:OMERO_USERDIR=(unset)
> Environment:OMERO_TMPDIR=(unset)
> Environment:PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/home/omero/OMERO.server/bin
> Environment:PYTHONPATH=(unset)
> Environment:ICE_HOME=(unset)
> Environment:LD_LIBRARY_PATH=(unset)
> Environment:DYLD_LIBRARY_PATH=(unset)
> 
> Cannot list deployed applications.
> OMERO data dir:'/OMERO_DATA/'                 Exists? True      Is writable? False
> OMERO temp dir:'/home/omero/omero/tmp'        Exists? True      Is writable? True   (Size: 0)
>

**Comment by joshmoore** [comment]
Hi @vncntprvst. Sorry for the slow response: we've just come back from break. Thanks for the diagnostics. I'm having a little trouble understanding what you've tried where. 

Reading the above, it looks to me like first you were hitting the "Ice version mismatch" which would be consistent with Ice 3.5 being installed. Then you have a "No module named Ice" which would be consistent with removing Ice 3.5.

Is it possible that Ice 3.6 is installed for a different Python installation? In the terminal where you see
```
In [1]: import Ice
In [2]: vers = Ice.stringVersion()
Out[3]: '3.6.5'
```
what does 

```
import sys
print sys.executable
print sys.path
```

show?

**Comment by vncntprvst** [comment]
Hi  Josh,
thanks for your reply. I did find what the issue was. During the installation, I had a problem with installing Ice 3.6 according to instruction (`pip install https://github.com/ome/zeroc-ice-py-ubuntu1604/releases/download/0.1.0/zeroc_ice-3.6.4-cp27-cp27mu-linux_x86_64.whl`) , which was telling me that the wheel file was not supported (turns out, all I needed to do was to upgrade pip ...). Instead, I had installed Ice with `pip install "zeroc-ice>3.5,<3.7" `. Once the wheel file issue was fixed, the rest worked. 
Just for completeness, here are the requested outputs:

> In [2]: print(sys.executable)
> /usr/bin/python3
> 
> In [3]: print(sys.path)
> ['/usr/local/bin', '/usr/lib/python35.zip', '/usr/lib/python3.5', '/usr/lib/python3.5/plat-x86_64-linux-gnu', '/usr/lib/python3.5/lib-dynload', '', '/usr/local/lib/python3.5/dist-packages', '/usr/lib/python3/dist-packages', '/usr/local/lib/python3.5/dist-packages/IPython/extensions', '/home/omero/.ipython']

Thanks
